org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/LocationManager.java

```diff
@@ -0,0 +1,184 @@
+import org.eclipse.scanning.api.points.IPointGenerator;
+import org.eclipse.scanning.api.points.IPosition;
+import org.eclipse.scanning.api.points.models.CompoundModel;
+import org.eclipse.scanning.api.scan.ScanningException;
+import org.eclipse.scanning.api.scan.models.ScanModel;
+
+/**
+ * 
+ * This class manages the location of
+ * various parts of the scan as the scan runs.
+ * 
+ * It maintains a count of levels and creates a 
+ * 
+ * @author Matthew Gerring
+ *
+ */
+public final class LocationManager {
+
+
+	/**
+	 * Variables used to monitor progress of inner scans
+	 */
+	private int outerSize  = 0;
+	private int outerCount = 0;
+	private int innerSize  = 0;
+	private int totalSize  = 0;
+	private int stepNumber = -1;
+	
+	// External data
+	private final ScanBean bean;
+	private final ScanModel model;
+	private final AnnotationManager manager;
+	
+	public LocationManager(ScanBean bean, ScanModel model, AnnotationManager manager) {
+		this.bean    = bean;
+		this.model   = model;
+		this.manager = manager;
+		manager.addDevices(this);
+	}
+	
+	public int getOuterSize() {
+		return outerSize;
+	}
+	public void setOuterSize(int outerSize) {
+		this.outerSize = outerSize;
+	}
+	public int getOuterCount() {
+		return outerCount;
+	}
+	public void setOuterCount(int outerCount) {
+		this.outerCount = outerCount;
+	}
+	public int getInnerSize() {
+		return innerSize;
+	}
+	public void setInnerSize(int innerSize) {
+		this.innerSize = innerSize;
+	}
+	public int getTotalSize() {
+		return totalSize;
+	}
+	public void setTotalSize(int totalSize) {
+		this.totalSize = totalSize;
+	}
+	public int getStepNumber() {
+		return stepNumber;
+	}
+	public void setStepNumber(int stepNumber) {
+		this.stepNumber = stepNumber;
+	}
+	
+	/**
+	 * Called during the scan to increment counts.
+	 */
+	@PointEnd
+	public void increment() {
+    	outerCount++;
+		stepNumber+=Math.max(innerSize, 1);
+	}
+
+	/**
+	 * Method used to generate an iterator for the scan.
+	 * It sets counts which are incremented during the scan.
+	 * 
+	 * @return
+	 * @throws ScanningException
+	 */
+	public Iterator<IPosition> createPositionIterator() throws ScanningException {
+		
+		CompoundModel<?> cmodel = bean.getScanRequest()!=null ? bean.getScanRequest().getCompoundModel() : null;
+		SubscanModerator moderator = new SubscanModerator(model.getPositionIterable(), cmodel, model.getDetectors(), ServiceHolder.getGeneratorService());
+		manager.addContext(moderator);
+		
+		try {
+			stepNumber = 0;
+			outerSize  = getEstimatedSize(moderator.getOuterIterable());
+			innerSize  = getEstimatedSize(moderator.getInnerIterable());
+			totalSize  = getEstimatedSize(model.getPositionIterable());
+		} catch (GeneratorException se) {
+			throw new ScanningException("Cannot create the position iterator!", se);
+		}
+
+		return moderator.getOuterIterable().iterator();
+	}
+
+
+	private int getEstimatedSize(Iterable<IPosition> gen) throws GeneratorException {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "getEstimatedSize" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)